### PR TITLE
KAFKA-7965: Fix testRollingBrokerRestartsWithSmallerMaxGroupSizeConfigDisruptsBigGroup

### PR DESCRIPTION
**Pull request by huxihx**
https://issues.apache.org/jira/browse/KAFKA-7965

Most of the time, the group coordinator runs on broker 1. Occasionally the group coordinator will be placed on broker 2. If that's the case, the loop starting at line 320 have no chance to check and update `kickedOutConsumerIdx`. A quick fix is to always restart the coordinator to ensure that `kickedOutConsumerIdx` could be checked after the new coordinator with new config gets to work.

*More detailed description of your change,
if necessary. The PR title and PR message become
the squashed commit message, so use a separate
comment to ping reviewers.*

*Summary of testing strategy (including rationale)
for the feature or bug fix. Unit and/or integration
tests are expected for any behaviour change and
system tests should be considered for larger changes.*

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by huxihx** [comment]
retest this please

**Comment by huxihx** [comment]
@bbejeck Please review this patch. Thanks.

**Comment by stanislavkozlovski** [comment]
> Occasionally the group coordinator will be placed on broker 2. If that's the case, the loop starting at line 320 have no chance to check and update kickedOutConsumerIdx

Why would the loop have no chance to check and update `kickedOutConsumerIdx`? If it is placed on broker 2 (last broker), the last iteration should call `killBroker(2)` and the pollers would hit an exception. Unless there is a race condition in between `killBroker(broker)` and the `consumerPollers(idx).thrownException match`, which is entirely possible I think.

This fix looks good to me. Would you mind running 100 iterations of the test with and without the changes to compare results?

File: core/src/test/scala/integration/kafka/api/ConsumerBounceTest.scala [hunk 1]
```diff
@@ -316,8 +316,14 @@ class ConsumerBounceTest extends AbstractConsumerTest with Logging {
     // roll all brokers with a lesser max group size to make sure coordinator has the new config
     val newConfigs = generateKafkaConfigs(maxGroupSize.toString)
     var kickedOutConsumerIdx: Option[Int] = None
+    val holdingGroupBrokers = servers.filter(!_.groupCoordinator.groupManager.currentGroups.isEmpty).map(_.config.brokerId)
+    // should only have one broker holding the group metadata
+    assertEquals(holdingGroupBrokers.size, 1)
+    val coordinator = holdingGroupBrokers.head
+    // ensure the coordinator broker will be restarted first
+    val orderedBrokersIds = List(coordinator) ++ servers.indices.toBuffer.filter(_ != coordinator)
```
Comment (hachikuji) [inline core/src/test/scala/integration/kafka/api/ConsumerBounceTest.scala:324]: I'm a little confused about the problem we're fixing. Why does it matter that the coordinator is restarted first? Overall, I wonder if we can simplify this. Our goal is to observe the effect of changing the max group size config. Could we restart all brokers first and then check the consumers for the exception we expect?

**Review by hachikuji — COMMENTED** [review]
Thanks @huxihx for looking at this failing test. Left one comment.

**Comment by hachikuji** [comment]
retest this please

**Comment by hachikuji** [comment]
retest this please

**Comment by hachikuji** [comment]
retest this please

**Review by hachikuji — APPROVED** [review]
LGTM. I will go ahead and merge to try and get this test stable. If the test continues to fail, we will can consider simplifying as I suggested.

**Pull request merged**